UnitySDK/Assets/ML-Agents/Scripts/Policy/BarracudaPolicy.cs

```diff
@@ -58,26 +62,26 @@ public void DecideAction()
         /// If this is the first Agent being checked, its Sensor sizes will be saved.
         /// </summary>
         /// <param name="agent">The Agent to check</param>
-        void ValidateAgentSensorShapes(Agent agent)
+        void ValidateAgentSensorShapes(List<Observation> observations)
```

Comment:
Can this just take the list of sensors instead? Feels wasteful to make it generate observations.

Reply:
Is it enough to validate the sensors? If there is an issue in the sensors generating the Observations, we will no longer catch it.
I can also try to make it static and take a list of sensors and a list of Shapes as input?

Reply:
(summarizing offline talk)

The old code isn't very useful since the Policy is 1:1 with the Agent. We'll move the shape checks to a helper class, and check in the model runner and communicator.